ZotBinsAPI.py

```diff
@@ -282,7 +289,94 @@ def get_obervation_stats():
+        if request.method == 'GET':
+            sensor_id = request.args.get("sensor_id")
+            start_timestamp = request.args.get("start_timestamp")
+            end_timestamp = request.args.get("end_timestamp")
+
+            """
+            Perform a quick null check and return
+            useful info to the caller so that they
+            know what fields they should be adding.
+            
+            TODO: Probably should have this null check 
+            anytim these params are being used.
+            """
+            if (sensor_id == None):
+                raise Exception("Param 'sensor_id' cannot be null")
+            if (start_timestamp == None):
+                raise Exception("Param 'start_timestamp' cannot be null")
+            if (end_timestamp == None):
+                raise Exception("Param 'end_timestamp' cannot be null")
+
+            obs_type = None
+            format_checker = re.compile("^(ZBin)(\d+)([a-zA-Z]+)$")
+
+            """
+            We can exactly parse the sensor_id string in to 3 groups:
+            (1) "ZBin", (2) sensor ID number, (3) sensor type.
+
+            Check if the formatting is correct too...
+            """
+            sensor_parts = format_checker.match(sensor_id)
+            if (sensor_parts == None):
+                raise Exception("Param 'sensor_id' must be in format 'ZBin[SENSOR_ID][SENSOR_TYPE]'")
+            sensor_groups = sensor_parts.groups()
+            obs_sensor_id = sensor_groups[1]
+            obs_id = 0
+            obs_type = sensor_groups[2]
+
+            random.seed(''.join([str(ord(c)) for c in sensor_id])) # so we always get same observations per sensor
 
+            for x in range(random.randint(10, 100)):
+                obs_dict = {"sensor_id" : obs_sensor_id, "id" : obs_id, "timestamp" : start_timestamp}
+                if (obs_type == 'B'):
+                    random_frequency_value = random.randint(0, 5000)
+                    obs_dict["payload"] = {"distance" : random_frequency_value}
+                elif (obs_type == 'D'):
+                    random_distance_value = random.randint(0, 100)
+                    obs_dict["payload"] = {"distance" : random_distance_value}
+                elif (obs_type == 'W'):
+                    random_weight_value = random.randint(0, 350)
+                    obs_dict["payload"] = {"distance" : random_weight_value}
+                elif (obs_type == 'Di'):
+                    random_diversion_value = random.randint(0, 100)
+                    obs_dict["payload"] = {"distance" : random_diversion_value}
+                else:
+                    raise Exception("Invalid sensor type... must be 'B', 'D', 'W', or 'Di'...")
+                ret.append(obs_dict)
+                obs_id += 1
+            return jsonify(ret)
+        # For robustness, added an else statement here...
+        else:
+            raise Exception("Must use GET method") 
+    except Exception as e:
+        print(e)
```

Comment:
If I understand correctly, this print statement doesn't really add any value since it will be printed to some console in pythonanywhere. More useful would be to write to some log file, but this is a mock API so it doesn't matter so much.

Reply:
Yes, agreed. I will leave this `print` for now.